src/status_im/models/wallet.cljs

```diff
@@ -51,3 +57,109 @@
+        contact (get contacts (utils.hex/normalize-hex to))]
+    (cond-> {:id               (str id)
+             :to-name          (or (when (nil? to)
+                                     (i18n/label :t/new-contract))
+                                   contact)
+             :symbol           :ETH
+             :method           method
+             :dapp-transaction queued-transaction
+
+             :to               to
+             ;; TODO (andrey) why don't we apply money/to-decimal for amount
+             :amount           (money/bignumber (or value 0))
+             :gas              (when (seq gas)
+                                 (money/bignumber (money/to-decimal gas)))
+             :gas-price        (when (seq gasPrice)
+                                 (money/bignumber (money/to-decimal gasPrice)))
+             :data             data}
+      nonce
+      (assoc :nonce nonce))))
+
+;; SEND TRANSACTION -> RPC TRANSACTION
+(defn prepare-send-transaction [from {:keys [amount to gas gas-price data nonce]}]
+  (cond-> {:from      (ethereum/normalized-address from)
+           :to        (ethereum/normalized-address to)
+           :value     (ethereum/int->hex amount)
+           :gas       (ethereum/int->hex gas)
+           :gas-price (ethereum/int->hex gas-price)}
+    data
+    (assoc :data data)
+    nonce
+    (assoc :nonce nonce)))
+
+(defn normalize-sign-message-params [params]
+  (let [first_param           (first params)
+        second_param          (second params)
+        first-param-address?  (ethereum/address? first_param)
+        second-param-address? (ethereum/address? second_param)]
+    (when (or first-param-address? second-param-address?)
+      (if first-param-address?
+        [first_param second_param]
+        [second_param first_param]))))
+
+(defn web3-error-callback [fx {:keys [webview-bridge]} {:keys [message-id]} message]
+  (assoc fx :send-to-bridge-fx [{:type      constants/web3-send-async-callback
+                                 :messageId message-id
+                                 :error     message}
+                                webview-bridge]))
+
+(defn dapp-complete-transaction [id result method message-id webview]
+  (cond-> {:send-to-bridge-fx [{:type      constants/web3-send-async-callback
+                                :messageId message-id
+                                :result    {:jsonrpc "2.0"
+                                            :id      (int id)
+                                            :result  result}}
+                               webview]
+           :dispatch          [:navigate-back]}
+
+    (= method constants/web3-send-transaction)
+    (assoc :dispatch-later [{:ms 400 :dispatch [:navigate-to-modal :wallet-transaction-sent-modal]}])))
+
+(defn discard-transaction
+  [{:keys [db]}]
+  (let [{:keys [dapp-transaction] :as transaction} (get-in db [:wallet :send-transaction])]
+    ;;TODO (andrey) why do we need here from-chat? (Goran?)
```

Comment:
@goranjovic Please comment

Reply:
It was a duct-tape fix for the two-step transaction signing flow, see #4924 PR.
Without that fix, all fields in Wallet Send would suddenly become editable if user taps Cancel or provides a wrong password. 

Since the first proper fix (dapp tx onboarding) is already merged in develop and this PR is the second proper fix, it's most likely unneeded now.

Reply:
This comment - https://github.com/status-im/status-react/pull/5455/files#diff-93d9c10db297821b5145f29bdbb7b6bcL141